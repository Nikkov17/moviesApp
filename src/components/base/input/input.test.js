import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Input from './input.component';

describe('input', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  describe('onFormSubmit', () => {
    it('should call updateMoviesList', () => {
      const component = shallow(
        <Input updateMoviesList={jest.fn()} />
      );
      const instance = component.instance();

      jest.spyOn(instance, 'onFormSubmit');
      instance.inputRef = {
        current: {
          value: 'check'
        }
      };

      component.find('.input-container-wrapper').simulate('submit', {
        preventDefault: jest.fn()
      });

      expect(instance.props.updateMoviesList).toHaveBeenCalledWith('check');
    });
  });

  describe('snapshot', () => {
    it('check snaphot', () => {
      const component = shallow(<Input updateMoviesList={() => {
        return true;
      }}
      />);
      expect(component).toMatchSnapshot();
    });
  });
});
