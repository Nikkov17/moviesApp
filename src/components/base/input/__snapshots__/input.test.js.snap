// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot check snaphot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    getMoviesList={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="input-container-wrapper"
        onSubmit={[Function]}
      >
        <label
          className="input-label"
          htmlFor="titleInput"
        >
          Find your movie
          <input
            className="input"
            id="titleInput"
            type="text"
          />
        </label>
        <button
          className="submit-button"
          type="submit"
        >
          Submit
        </button>
      </form>,
      "className": "input-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="input-label"
            htmlFor="titleInput"
          >
            Find your movie
            <input
              className="input"
              id="titleInput"
              type="text"
            />
          </label>,
          <button
            className="submit-button"
            type="submit"
          >
            Submit
          </button>,
        ],
        "className": "input-container-wrapper",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Find your movie",
              <input
                className="input"
                id="titleInput"
                type="text"
              />,
            ],
            "className": "input-label",
            "htmlFor": "titleInput",
          },
          "ref": null,
          "rendered": Array [
            "Find your movie",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input",
                "id": "titleInput",
                "type": "text",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "submit-button",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="input-container-wrapper"
          onSubmit={[Function]}
        >
          <label
            className="input-label"
            htmlFor="titleInput"
          >
            Find your movie
            <input
              className="input"
              id="titleInput"
              type="text"
            />
          </label>
          <button
            className="submit-button"
            type="submit"
          >
            Submit
          </button>
        </form>,
        "className": "input-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="input-label"
              htmlFor="titleInput"
            >
              Find your movie
              <input
                className="input"
                id="titleInput"
                type="text"
              />
            </label>,
            <button
              className="submit-button"
              type="submit"
            >
              Submit
            </button>,
          ],
          "className": "input-container-wrapper",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Find your movie",
                <input
                  className="input"
                  id="titleInput"
                  type="text"
                />,
              ],
              "className": "input-label",
              "htmlFor": "titleInput",
            },
            "ref": null,
            "rendered": Array [
              "Find your movie",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "id": "titleInput",
                  "type": "text",
                },
                "ref": Object {
                  "current": null,
                },
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "submit-button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
